pkg/backup/util/aws_ebs.go

```diff
@@ -144,46 +146,52 @@ func NewEC2Session(concurrency uint) (*EC2Session, error) {
 func (e *EC2Session) DeleteSnapshots(snapIDMap map[string]string) error {
 	var (
 		deletedCnt atomic.Int32
-		errs       error
-		lock       sync.Mutex
 	)
 
 	eg, _ := errgroup.WithContext(context.Background())
 	workerPool := NewWorkerPool(e.concurrency, "delete snapshots")
 	for volID := range snapIDMap {
 		snapID := snapIDMap[volID]
 		workerPool.ApplyOnErrorGroup(eg, func() error {
-			_, err := e.EC2.DeleteSnapshot(&ec2.DeleteSnapshotInput{
-				SnapshotId: &snapID,
-			})
-			if err != nil {
-				if strings.Contains(err.Error(), "NotFound") {
-					klog.Warningf("snapshot %s not found, err: %s", snapID, err.Error())
+			// use exponential backoff, every retry duration is duration * factor ^ (used_step - 1)
+			backoff := wait.Backoff{
+				Duration: time.Second,
+				Steps:    8,
+				Factor:   2.0,
+				Cap:      time.Minute,
+			}
+			delSnapshots := func() error {
+				_, err := e.EC2.DeleteSnapshot(&ec2.DeleteSnapshotInput{
+					SnapshotId: &snapID,
+				})
+				if err != nil {
+					if strings.Contains(err.Error(), "NotFound") {
```

Comment:
Perhaps handle the error by code like:
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/handling-errors.html